ga4gh/backend.py

```diff
@@ -413,6 +428,24 @@ def getVariantSet(self, id_):
         dataset = self._getDatasetFromCompoundId(id_)
         return self.runGetRequest(dataset.getVariantSetIdMap(), id_)
 
+    def getVariant(self, id_):
+        """
+        Returns a variant with the given id
+        """
+        try:
+            splits = id_.split(':')
+        except AttributeError:
+            raise exceptions.BadIdentifierException(id_)
+        try:
+            variantSetId = '{}:{}'.format(splits[0], splits[1])
+        except KeyError:
+            raise exceptions.BadIdentifierException(id_)
+        dataset = self._getDatasetFromCompoundId(id_)
+        variantSet = _safeMapQuery(
+            dataset.getVariantSetIdMap(), variantSetId)
+        variantId = ':'.join(splits[2:])
+        raise Exception(variantSet + variantId)  # TODO
```

Comment:
Good questions @dcolligan! The ID field in the VCF corresponds to the `name` field in the GA4GH objects, and it's basically useless as an identifier.  As for our ID, I think we'll have to do something like ```datasetId:variantSetId:chrom:pos:md5(REF + ALT)``. The last part is necessary because we can have many variants with the same sort coordinate, and the only way to be sure it's _this_ exact variant  is to use the REF and ALT sequences. I think it's a good idea to use md5 here (or some other hash) because our IDs could become arbitrarily long otherwise.

In terms of finding the variant in question, we can use pysam to seek to the first record overlapping (chrom,pos) and then iterate from there until we hit the VCF record we're interested in. This will be fast the majority of the time.